src/tools/pkcs11-tool.c

```diff
@@ -4575,6 +4594,10 @@ static CK_RV write_object(CK_SESSION_HANDLE session)
 	EVP_PKEY_free(evp_key);
 #endif /* ENABLE_OPENSSL */
 
+	if (contents)
+		free(contents);
+	if (certdata)
+		free(certdata);
```

Comment:
nit: No need to check for null here. If it is null, the `free()` is no-op anyway (and if it was freed or is uninitialized you are screwed anyway).
```suggestion
	free(contents);
	free(certdata);
```